src/autoqasm/simulator/program_context.py

```diff
@@ -163,7 +163,7 @@ def _get_indices_length(
         elif isinstance(last_index, DiscreteSet):
             return len(last_index.values)
         else:
-            raise TypeError(f"tuple indices must be integers or slices, not {type(last_index)}")
+            raise TypeError(f"slice indices must be integers, not {type(last_index)}")
```

Comment:
I think this change makes the error message too specific; we can hit this error for any invalid index, it's not necessarily due to an invalid slice. For example, in addition to invalid slices like `q[1:0.1]`, we would also hit this for any invalid array index, e.g. `q["abc"]` or `q[1.23]`.

What's the type of `last_index` here in the case you have in the description, where you have an invalid slice? Could we add another `elif` clauses here specifically for that type?

Reply:
I changed still include the error message. But separate the validation paths to different test cases.